app/forms/mappings/LocalDateFormatter.scala

```diff
@@ -103,7 +102,15 @@ private[mappings] class LocalDateFormatter(invalidKey: String,
   override def bind(key: String, data: Map[String, String]): Either[Seq[FormError], LocalDate] = {
     val fields = fieldKeys.map {
       field =>
-        field -> data.get(s"$key.$field").filter(_.nonEmpty)
+        field -> {
+          val processingData = data.get(s"$key.$field").filter(_.nonEmpty)
+          if (processingData.isDefined) {
+            val dataWithoutSpaces = processingData.get.replaceAll(" ", "")
+            if(dataWithoutSpaces.isEmpty) None else Some(dataWithoutSpaces)
+          } else {
+            processingData
+          }
```

Comment:
This seems a bit overly complex, could you not just have:
```suggestion
          data.get(s"$key.$field").map(_.replaceAll(" ", "")).filter(_.nonEmpty)
```

Reply:
Huh, I did try something similar but I think I wasn't doing it properly! Will change to this